public/app/plugins/panel/geomap/editor/MapViewEditor.tsx

```diff
@@ -0,0 +1,120 @@
+  context,
+}) => {
+  const labelWidth = 10;
+
+  const views = useMemo(() => {
+    const ids: string[] = [];
+    if (value?.id) {
+      ids.push(value.id);
+    } else {
+      ids.push(centerPointRegistry.list()[0].id);
+    }
+    return centerPointRegistry.selectOptions(ids);
+  }, [value?.id]);
+
+  const onSetCurrentView = useCallback(() => {
+    const map = lastGeomapPanelInstance?.map;
+    if (map) {
+      const view = map.getView();
+      const coords = view.getCenter();
+      if (coords) {
+        const center = toLonLat(coords, view.getProjection());
+        onChange({
+          ...value,
+          id: MapCenterID.Coordinates,
+          lon: +center[0].toFixed(6),
+          lat: +center[1].toFixed(6),
+          zoom: +view.getZoom()!.toFixed(2),
+        });
+      }
+    }
+  }, [value, onChange]);
+
+  const onSelectView = useCallback(
+    (selection: SelectableValue<string>) => {
+      const v = centerPointRegistry.getIfExists(selection.value);
+      if (v) {
+        onChange({
+          ...value,
+          id: v.id,
+          lat: v.lat ?? value.lat,
+          lon: v.lon ?? value.lon,
+          zoom: v.zoom ?? value.zoom,
+        });
+      }
+    },
+    [value, onChange]
+  );
+
+  return (
+    <>
+      <InlineFieldRow>
+        <InlineField label="View" labelWidth={labelWidth} grow={true}>
+          <Select options={views.options} value={views.current} onChange={onSelectView} />
+        </InlineField>
+      </InlineFieldRow>
+      {value?.id === MapCenterID.Coordinates && (
+        <>
+          <InlineFieldRow>
+            <InlineField label="Latitude" labelWidth={labelWidth} grow={true}>
+              <NumberInput
+                value={value.lat}
+                min={-90}
+                max={90}
+                step={0.001}
+                onChange={(v) => {
+                  onChange({ ...value, lat: v });
+                }}
+              />
+            </InlineField>
+          </InlineFieldRow>
+          <InlineFieldRow>
+            <InlineField label="Longitude" labelWidth={labelWidth} grow={true}>
+              <NumberInput
+                value={value.lon}
+                min={-180}
+                max={180}
+                step={0.001}
+                onChange={(v) => {
+                  onChange({ ...value, lon: v });
+                }}
+              />
+            </InlineField>
+          </InlineFieldRow>
+        </>
+      )}
+
+      <InlineFieldRow>
+        <InlineField label="Zoom" labelWidth={labelWidth} grow={true}>
+          <NumberInput
+            value={value.zoom ?? 1}
+            min={1}
+            max={18}
+            step={0.01}
+            onChange={(v) => {
+              onChange({ ...value, zoom: v });
+            }}
+          />
+        </InlineField>
+      </InlineFieldRow>
+
+      <VerticalGroup>
+        <Button variant="secondary" size="sm" fullWidth onClick={onSetCurrentView}>
+          <span>Use current view</span>
```

Comment:
as someone who's not really familiar with the world map panel, it wasn't immediately obvious what `Use current view` meant. i think it means that next time you load the panel, it will display at these coords. if so, i wonder if something like `Set as default view` would be better?

Reply:
good point -- I changed it to:  
![image](https://user-images.githubusercontent.com/705951/126182302-3a1abd48-0499-4ec1-a732-ec8a7576acbe.png)

and will be talking with the docs team and @pkolyvas tomorrow :)